assets/src/components/v2/gl_eink_double/line_map.tsx

```diff
@@ -0,0 +1,418 @@
+
+const BaseMapFutureTerminal = () => {
+  return (
+    <circle
+      cx={LEFT_MARGIN + LINE_WIDTH / 2}
+      cy={TOP_MARGIN}
+      r={BIG_STOP_RADIUS}
+      className="line-map--future"
+    />
+  );
+};
+
+const BaseMapFutureArrow = () => {
+  const arrowHeight = LINE_WIDTH / 2;
+  const lineHeight = 72;
+
+  const pathData = [
+    "M",
+    LEFT_MARGIN,
+    TOP_MARGIN,
+    "L",
+    LEFT_MARGIN,
+    TOP_MARGIN - lineHeight,
+    "L",
+    LEFT_MARGIN + LINE_WIDTH / 2,
+    TOP_MARGIN - (lineHeight + arrowHeight),
+    "L",
+    LEFT_MARGIN + LINE_WIDTH,
+    TOP_MARGIN - lineHeight,
+    "L",
+    LEFT_MARGIN + LINE_WIDTH,
+    TOP_MARGIN,
+    "Z",
+  ].join(" ");
+
+  return <path d={pathData} className="line-map--future" />;
+};
+
+const BaseMapFuture = ({ stops }) => {
+  const numFutureStops = stops.findIndex((stop) => stop.current);
+
+  return (
+    <>
+      <rect
+        x={LEFT_MARGIN}
+        y={TOP_MARGIN}
+        width={LINE_WIDTH}
+        height={STOP_SPACING * numFutureStops}
+        className="line-map--future"
+      />
+      {stops[0].terminal ? <BaseMapFutureTerminal /> : <BaseMapFutureArrow />}
+    </>
+  );
+};
+
+const BaseMapCurrent = ({ stops }) => {
+  const numFutureStops = stops.findIndex((stop) => stop.current);
+
+  return (
+    <circle
+      cx={LEFT_MARGIN + LINE_WIDTH / 2}
+      cy={TOP_MARGIN + STOP_SPACING * numFutureStops}
+      r={BIG_STOP_RADIUS}
+      className="line-map--future"
+    />
+  );
+};
+
+const BaseMapPast = ({ stops }) => {
+  const numStops = stops.length;
+  const numFutureStops = stops.findIndex((stop) => stop.current);
+  const numPastStops = numStops - (numFutureStops + 1);
+
+  let height;
+  if (stops[stops.length - 1].terminal) {
+    height = STOP_SPACING * numPastStops;
+  } else {
+    height = HEIGHT - (TOP_MARGIN + STOP_SPACING * numFutureStops);
+  }
+
+  return (
+    <>
+      <rect
+        x={LEFT_MARGIN}
+        y={TOP_MARGIN + STOP_SPACING * numFutureStops}
+        width={LINE_WIDTH}
+        height={height}
+        className="line-map--past"
+      />
+
+      {stops[stops.length - 1].terminal && (
+        <circle
+          cx={LEFT_MARGIN + LINE_WIDTH / 2}
+          cy={TOP_MARGIN + STOP_SPACING * (numStops - 1)}
+          r={BIG_STOP_RADIUS}
+          className="line-map--past"
+        />
+      )}
+    </>
+  );
+};
+
+const BaseMapStops = ({ stops }) => {
+  return (
+    <>
+      {stops.map(({ label, downstream, current, terminal }, i) => {
+        const stopIcon = (
+          <circle
+            cx={LEFT_MARGIN + LINE_WIDTH / 2}
+            cy={TOP_MARGIN + STOP_SPACING * i}
+            r={STOP_RADIUS}
+            className="line-map__stop"
+          />
+        );
+
+        let stopLabel = null;
+        if (downstream || current || terminal) {
+          let modifier;
+          if (current) {
+            modifier = "current";
+          } else if (!downstream && !current) {
+            modifier = "past";
+          }
```

Comment:
We don't currently have any e-Ink signs where the end of the route appears on the line map, since we don't have any within two stops of the terminal. I'm not sure how we'd want to display that.

For terminals in the past, we just want `modifier = "past"`.